File: meditrack-backend/src/main/java/com/meditrack/meditrack_backend/models/MedicineRequest.java
package com.meditrack.meditrack_backend.models;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Data
@Document(collection = "medicineRequests")
public class MedicineRequest {
    @Id
    private String id;
    private String doctorId;
    private String medicineName;
    private String medicineType;
    private String dosage;
    private Integer quantity;
    private String reason;
    private String priorityLevel;
    private String status; // "Pending", "Approved", "Rejected"
    private Date requestedAt;
}

File: lib/models/MedicineRequest.ts
import mongoose from 'mongoose';

const medicineRequestSchema = new mongoose.Schema(
  {
    doctorId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'DoctorProfile',
      required: true,
    },
    medicineName: {
      type: String,
      required: true,
    },
    dosage: {
      type: String,
      required: true,
    },
    reason: {
      type: String,
      required: true,
    },
    status: {
      type: String,
      enum: ['pending', 'approved', 'rejected'],
      default: 'pending',
    },
    adminNotes: String,
  },
  { timestamps: true }
);

export const MedicineRequest = mongoose.models.MedicineRequest || mongoose.model('MedicineRequest', medicineRequestSchema);
